### PR TITLE
reenable patch serverside using strategic-merge-patch

### PR DESCRIPTION
**Pull request by mikedanese**
```
➜  kubernetes git:(re-patch) kubectl update node e2e-test-mikedanese-minion-0rfr --patch='{"spec":{"unschedulable":true}}'
e2e-test-mikedanese-minion-0rfr
➜  kubernetes git:(re-patch) kubectl get node e2e-test-mikedanese-minion-0rfr                                       
NAME                              LABELS                                                   STATUS
e2e-test-mikedanese-minion-0rfr   kubernetes.io/hostname=e2e-test-mikedanese-minion-0rfr   Ready,SchedulingDisabled
```

### CODE REVIEW

File: pkg/kubectl/resource/helper.go [hunk 1]
```diff
@@ -43,11 +44,10 @@ type Helper struct {
...
-		Resource:   mapping.Resource,
-		Codec:      mapping.Codec,
-		Versioner:  mapping.MetadataAccessor,
-
+		RESTClient:      client,
+		Resource:        mapping.Resource,
+		Codec:           mapping.Codec,
+		Versioner:       mapping.MetadataAccessor,
 		NamespaceScoped: mapping.Scope.Name() == meta.RESTScopeNameNamespace,
 	}
```
Comment (mikedanese) [inline pkg/kubectl/resource/helper.go]: deleted a weird newline in the middle of this struct initializer

**Comment by bgrant0607** [comment]
Cool, thanks. Do we have a test for --patch? Does it work for the specific use case cited in the issue?

cc @jlowdermilk @ghodss

**Comment by mikedanese** [comment]
I'm currently adding a test to ./hack/test-cmd.sh but I'm going to look for other spots to plugin and write unit tests.

File: docs/kubectl_update.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ $ cat pod.json | kubectl update -f -
 ```
   -f, --filename=[]: Filename, directory, or URL to file to use to update the resource.
   -h, --help=false: help for update
+      --patch="": A JSON document to override the existing resource. The resource is downloaded, patched with the JSON, then updated.
 ```
```
Comment (jlowdermilk) [inline docs/kubectl_update.md]: Can you re-add a `--patch` example? Either marking a node unschedulable or updating the image are reasonable ones.

**Comment by mikedanese** [comment]
@bgrant0607, @jlowdermilk has appened a commit to re-enable patch tests in ./hack/test-cmd.sh

**Comment by jlowdermilk** [comment]
LGTM, other than re-adding an example to the help text.

**Comment by bgrant0607** [comment]
LGTM

cc @eparis

**Comment by bgrant0607** [comment]
+1 to an example

File: docs/kubectl_update.md (hunk 2 above)
Comment (mikedanese) [inline docs/kubectl_update.md]: @jlowdermilk Done.

File: docs/kubectl_update.md [hunk 3]
```diff
@@ -21,6 +21,10 @@ $ kubectl update -f pod.json
 
 // Update a pod based on the JSON passed into stdin.
 $ cat pod.json | kubectl update -f -
+
+// Partially update a pod using strategic merge patch
```
Comment (jlowdermilk) [inline docs/kubectl_update.md]: Mismatch between description and command; command is updating node. Also, mentioning/linking to strategic merge patch is overkill imo. Users don't need to know the implementation detail, as cool as it is.

Comment (mikedanese) [inline docs/kubectl_update.md]: whoops, fixed.

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by jlowdermilk** [comment]
LGTM. Thanks, Mike!

**Pull request merged**